pkgs/development/cuda-modules/cuda-library-samples/extension.nix

```diff
@@ -1,5 +1,12 @@
-{ hostPlatform, lib }:
+{
+  lib,
+  # NOTE: We would use backendStdenv to get the hostPlatform, but this file is called by pkgs.callPackage.
+  # This is an exception situation where we can't use backendStdenv to get the hostPlatform.
```

Comment:
After a bit more of thinking. I don't see any issue with taking `hostPlatform` from `stdenv` instead of `backendStdenv`. The mere purpose of the latter is to reconfigure the cc-wrapper.

Maybe this comment belongs in `cuda-packages.nix` where we import all the `extension.nix`-s. And then we don't have to repeat it all over the place either.

Also what is the original reason for the change? The reason we have to take `hostPlatform` from a stdenv, rather than from the scope

> an exception situation

A typo, also maybe "temporary" is better after all

Reply:
Sure, I'll change those. And good catch on the typo!

See https://github.com/NixOS/nixpkgs/pull/45717 and https://github.com/NixOS/nixpkgs/issues/27069 for the motivation behind these changes.

Reply:
Good point as well about the difference between `backendStdenv` and `stdenv` -- I'll switch it all over to `stdenv`.

Reply:
Fixed in latest